src/main/java/com/example/Restaurant_Management_System_REST_API/service/CustomerService.java

```diff
@@ -137,4 +139,15 @@ public ResponseEntity<?> delete(Long id) throws NotFoundInDatabaseException {
         return new ResponseEntity<>("Customer: " + customerToDelete.getUsername() + " has been successfully deleted!",
                 HttpStatus.OK);
     }
+
+    Optional<Customer> getCustomerFromReservationByEmailAddress(Reservation reservation) {
+        String emailAddress = reservation.getCustomer().getEmailAddress();
+        return customerRepository.findByEmailAddress(emailAddress);
+    }
+
+    void checkIfCustomerHasAnyReservation(Customer customer) throws CustomerAlreadyHasReservationException {
+        if (customer.getReservation() != null && customer.getReservation().getId() != null) {
+            throw new CustomerAlreadyHasReservationException();
+        }
+    }
```

Comment:
yes, I wanted that these methods are accessible from the classes in the same package -> they are used in ReservationService and ReservationServiceTest classes

Reply:
ReservationServiceTest is not in the same package, so it can't use those methods. And in the ReservationService you are using those through the CustomerService object anyway, so you can simply set them to private

Reply:
I cannot
![image](https://github.com/BartoliniAndBorderCollies/Restaurant_Management_System_REST_API/assets/126821059/c4b0572c-3311-4cd8-8f30-b4b12a8b1ec8)

![image](https://github.com/BartoliniAndBorderCollies/Restaurant_Management_System_REST_API/assets/126821059/2cc7b507-884d-4c55-9a49-5ecc7545d8a3)

Reply:
![image](https://github.com/BartoliniAndBorderCollies/Restaurant_Management_System_REST_API/assets/126821059/e71bb345-948e-4514-a869-bdfaefdf1c91)

![image](https://github.com/BartoliniAndBorderCollies/Restaurant_Management_System_REST_API/assets/126821059/a05fb6b3-3cba-4647-b085-1905797050c2)

Reply:
Okay, I see my mistake now. Default accessor to access from same package is okay!